support-frontend/assets/pages/contributions-landing/components/ContributionAmount.jsx

```diff
@@ -126,21 +126,17 @@ function withProps(props: PropTypes) {
     <ContributionTextInputDs
       id="contributionOther"
       name="contribution-other-amount"
-      type="number"
       label={`Other amount (${otherLabelSymbol})`}
       value={otherAmount}
-      onInput={e => updateAmount(
-      (e.target: any).value,
-      props.countryGroupId,
-      props.contributionType,
-    )}
-      isValid={checkOtherAmount(otherAmount || '', props.countryGroupId, props.contributionType)}
-      formHasBeenSubmitted={(checkoutFormHasBeenSubmitted || stripePaymentRequestButtonClicked)}
+      onInput={e => props.updateOtherAmount(
+        (e.target: any).value,
+        props.countryGroupId,
+        props.contributionType,
+      )}
+      isValid={checkAmount(otherAmount || '', props.countryGroupId, props.contributionType)}
+      formHasBeenSubmitted={(checkoutFormHasBeenSubmitted || stripePaymentRequestButtonClicked || otherAmount)}
```

Comment:
The generic `ContributionTextInputDs` component does not display errors until `formHasBeenSubmitted`.
But for this particular field we want to display errors as soon as `otherAmount` has a value.

Reply:
I guess this is maybe a bad prop name? The input shouldn't really care if the form has been submitted or not, just whether or not it should show an error message. Perhaps (and this can be done as a follow up) we should consider renaming to e.g `shoudShowErrorMessage`?

Reply:
Yes I wondered this. We currently have these props:
```
errorMessage: string | null,
isValid: boolean,
formHasBeenSubmitted: boolean,
```

Perhaps this would be better?
```
errorMessage: string | null,
shouldShowErrorMessage: boolean,
```

Reply:
Yeah I think that would definitely be an improvement. In fact I'm not 100% if we need this wrapper around the design systems text input?

Reply:
The only reason for it appears to be the `cssInvalid` override, but I just tested removing it and it makes no difference!

Reply:
I've just pushed a commit which deletes this component as it wasn't adding much.
And setting the error is now as simple as passing `string | null` to the `TextInput`